src/game/Tactical/UI_Cursors.cc

```diff
@@ -241,6 +244,26 @@ static UICursorID HandleActivatedTargetCursor(SOLDIERTYPE* const s, GridNo const
 		gfCannotGetThrough = chance < OK_CHANCE_TO_GET_THROUGH;
 	}
 
+	if(gamepolicy(show_hit_chance))
+	{
+		// Calculate chance to hit
+		if (recalc || giLastBodyLocationTargeted != s->bAimShotLocation || giLastAimTime != s->bShownAimTime)
+		{
+			GridNo targetTile = gUIFullTarget ? gUIFullTarget->sGridNo : map_pos;
+			giHitChance = is_throwing_knife ? CalcThrownChanceToHit(s, targetTile, s->bShownAimTime / 2, s->bAimShotLocation) :
+				CalcChanceToHitGun(s, targetTile, s->bShownAimTime / 2, s->bAimShotLocation, false);
+			giHitChance *= SoldierToLocationChanceToGetThrough(s, targetTile, s->bTargetLevel, s->bTargetCubeLevel, 0) / 100.0f;
+		}
+	
+		// Attach chance-to-hit to mouse cursor
+		if(giHitChance != -1)
+		{
+			SetChanceToHitText(st_format_printf("%d%%", giHitChance));
```

Comment:
Since you asked: there are also the native ST variants that just use `{}` for placeholders when you don't need any modifiers. Random example:
`ST::format("Invalid mineId {}", lair->associatedMineId);`